From 43934fbbf16c7f19e07ae1de17fce6697fb137cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADt=20Ondruch?= <vondruch@redhat.com>
Date: Tue, 8 Nov 2016 16:25:24 +0100
Subject: [PATCH] Use primary group as well as supplementary groups.

I might happen in certain environments (systemd-nspawn) that process has
no supplementary groups, but primary groups should be enough to pass most
of the tests.
---
 test/fileutils/test_fileutils.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/fileutils/test_fileutils.rb b/test/fileutils/test_fileutils.rb
index 0ff0aad..fa5b24d 100644
--- a/test/fileutils/test_fileutils.rb
+++ b/test/fileutils/test_fileutils.rb
@@ -142,7 +142,7 @@ def mymkdir(path)
 
   def setup
     @prevdir = Dir.pwd
-    @groups = Process.groups if have_file_perm?
+    @groups = [Process.gid, Process.groups].flatten if have_file_perm?
     tmproot = @tmproot = Dir.mktmpdir "fileutils"
     Dir.chdir tmproot
     my_rm_rf 'data'; mymkdir 'data'
-- 
2.10.1

